sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -175,19 +180,39 @@ def test_check_estimator():
 
     # doesn't error on actual estimator
     check_estimator(AdaBoostClassifier)
+    check_estimator(AdaBoostClassifier())
     check_estimator(MultiTaskElasticNet)
+    check_estimator(MultiTaskElasticNet())
+
+
+def test_check_estimator_clones():
+    # check that check_estimator doesn't modify the estimator it receives
+    from sklearn.datasets import load_iris
+    iris = load_iris()
+    est = AdaBoostClassifier()
```

Comment:
also: this is basically impossible without estimator tags right now, or we need to replicate the skipping logic etc here... hm